### PR TITLE
sinkv2(ticdc): use table sink state instead of table state

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/5928

### What is changed and how it works?

use table sink state instead of table state. This will simplify the implementation. After this, we will not need to move the table state into the table sink and keep it in the sink node.

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

**Review by asddongmen — APPROVED** [review]

File: cdc/sinkv2/eventsink/mq/worker.go [hunk 1]
```diff
@@ -169,7 +169,7 @@ func (w *worker) asyncSend(
 	for key, events := range partitionedRows {
 		for _, event := range events {
 			// Skip this event when the table is stopping.
-			if event.TableStatus.Load() == pipeline.TableStateStopping {
+			if event.SinkState.Load() == state.TableSinkStopping {
```
Comment (liuzix) [inline cdc/sinkv2/eventsink/mq/worker.go:172]: Can you wrap `event.SinkState.Load()` in a method, probably called `event.GetTableState()`?

File: cdc/sinkv2/tablesink/state/state.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+package state
+
+import "sync/atomic"
+
+// TableSinkState is state of the table sink.
+type TableSinkState int32
+
+// State for table sink
+const (
+	TableSinkStateUnknown TableSinkState = iota
```
Comment (liuzix) [inline cdc/sinkv2/tablesink/state/state.go:23]:
```suggestion
	TableSinkStateUnknown TableSinkState = iota+1
```
Make zero values panic.

File: cdc/sinkv2/eventsink/mq/worker.go (hunk 1 above)
Comment (hi-rustin) [reply cdc/sinkv2/eventsink/mq/worker.go:172]: Good suggestion!

File: cdc/sinkv2/tablesink/state/state.go (hunk 2 above)
Comment (hi-rustin) [reply cdc/sinkv2/tablesink/state/state.go:23]: Is this a best practice? Why do we need this? Could you please explain a little bit?

**Review by nongfushanquan — APPROVED** [review]

**Comment by hi-rustin** [comment]
/merge

@liuzix, I'm gonna merge this PR. If you have any other comments, please feel free to `/hold` this PR.

**Pull request merged**